xonsh/environ.py

```diff
@@ -891,6 +912,7 @@ def current_branch(pad=True):
         branch = get_hg_branch()
     if isinstance(branch, subprocess.TimeoutExpired):
         branch = '<branch-timeout>'
+        _first_branch_timeout_message()
```

Comment:
Is there a way to have it time out on the color _if we found a branch_?  Sorry to make more complications for you, but that seems to more likely place to timeout, so I would like to have the error message when the coloring fails, if we can.